packages/cli/src/config.ts

```diff
@@ -2,45 +2,95 @@ import { promises as fs } from 'fs';
 import * as yaml from 'js-yaml';
 import * as os from 'os';
 import * as path from 'path';
+import { v4 as uuidv4 } from 'uuid';
 
 const configDir = path.join(os.homedir(), '.config', 'lightdash');
 const configFilePath = path.join(configDir, 'config.yaml');
 
-type Config = {
+export type Config = {
+    user?: {
+        userUuid?: string;
+        anonymousUuid?: string;
+    };
     context?: {
         serverUrl?: string;
         project?: string;
         apiKey?: string;
     };
 };
 
-export const getConfig = async (useEnv: boolean = true): Promise<Config> => {
-    let validated: Config;
+export const setConfig = async (config: Config) => {
+    await fs.mkdir(path.dirname(configFilePath), { recursive: true });
+    await fs.writeFile(configFilePath, yaml.dump(config), 'utf8');
+};
+
+const getRawConfig = async (): Promise<Config> => {
     try {
         const raw = yaml.load(await fs.readFile(configFilePath, 'utf8'));
-        validated = raw as Config;
+        return raw as Config;
     } catch (e) {
         if (e.code === 'ENOENT') {
-            validated = {};
-        } else {
-            throw e;
+            return {} as Config;
         }
+        throw e;
+    }
+};
+
+const setAnonymousUuid = async (anonymousUuid: string) => {
+    const config = await getRawConfig();
+    await setConfig({
+        ...config,
+        user: {
+            ...(config.user || {}),
+            anonymousUuid,
+        },
+    });
+};
+
+export const getConfig = async (): Promise<Config> => {
+    const rawConfig = await getRawConfig();
+    if (rawConfig.user?.anonymousUuid === undefined) {
+        await setAnonymousUuid(uuidv4());
```

Comment:
Should we have `return getConfig();` to get the saved config with the anonymous uuid